Core.ApplicationServices/Organizations/StsOrganizationSynchronizationService.cs

```diff
@@ -149,18 +156,32 @@ public Maybe<OperationError> UpdateConnection(Guid organizationId, Maybe<int> le
                     {
                         if (consequences.DeletedExternalUnitsBeingDeleted.Any())
                         {
-                            _organizationUnitRepository.RemoveRange(consequences.DeletedExternalUnitsBeingDeleted);
+                            _organizationUnitRepository.RemoveRange(consequences.DeletedExternalUnitsBeingDeleted.Select(x => x.organizationUnit).ToList());
                         }
                         foreach (var (affectedUnit, _, _) in consequences.OrganizationUnitsBeingRenamed)
                         {
                             _domainEvents.Raise(new EntityUpdatedEvent<OrganizationUnit>(affectedUnit));
                         }
                         return consequences;
                     })
+                    .Bind(consequences =>
+                    {
+                        var error = LogChanges(organization.StsOrganizationConnection, consequences);
```

Comment:
should also be called in the "CreateConnection" scenario. 
In that case you can provide the consequences in the app service by extracting all fk org units and setting them as "added" in the consequences ctor...
You can use `organization.GetRoot().Flatten().Where(x=>x.origin==....)` to get the fk org units